lib/openstudio-standards/schedules/information.rb

```diff
@@ -40,14 +374,232 @@ def self.schedule_ruleset_get_min_max(schedule_ruleset)
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Schedules.Information', 'Year description is not specified. Full load hours calculation will assume 2009, the default year OS uses.')
+        year_start_date = OpenStudio::Date.new(OpenStudio::MonthOfYear.new('January'), 1, 2009)
+        year_end_date = OpenStudio::Date.new(OpenStudio::MonthOfYear.new('December'), 31, 2009)
+      end
+
+      # Get the ordered list of all the day schedules
+      day_schs = schedule_ruleset.getDaySchedules(year_start_date, year_end_date)
+
+      # Get the array of which schedule is used on each day of the year
+      day_schs_used_each_day = schedule_ruleset.getActiveRuleIndices(year_start_date, year_end_date)
+
+      # Create a map that shows how many days each schedule is used
+      day_sch_freq = day_schs_used_each_day.group_by { |n| n }
+
+      # Build a hash that maps schedule day index to schedule day
+      schedule_index_to_day = {}
+      day_schs.each_with_index do |day_sch, i|
+        schedule_index_to_day[day_schs_used_each_day[i]] = day_sch
+      end
+
+      # Loop through each of the schedules that is used, figure out the
+      # full load hours for that day, then multiply this by the number
+      # of days that day schedule applies and add this to the total.
+      annual_flh = 0.0
+      max_daily_flh = 0.0
+      default_day_sch = schedule_ruleset.defaultDaySchedule
+      day_sch_freq.each do |freq|
+        sch_index = freq[0]
+        number_of_days_sch_used = freq[1].size
+
+        # Get the day schedule at this index
+        day_sch = nil
+        day_sch = if sch_index == -1 # If index = -1, this day uses the default day schedule (not a rule)
+                    default_day_sch
+                  else
+                    schedule_index_to_day[sch_index]
+                  end
+        daily_flh = OpenstudioStandards::Schedules.schedule_day_get_equivalent_full_load_hours(day_sch)
+
+        # Multiply the daily EFLH by the number
+        # of days this schedule is used per year
+        # and add this to the overall total
+        annual_flh += daily_flh * number_of_days_sch_used
+      end
+
+      # Warn if the max daily EFLH is more than 24,
+      # which would indicate that this isn't a fractional schedule.
+      if max_daily_flh > 24
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Schedules.Information', "#{schedule_ruleset.name} has more than 24 EFLH in one day schedule, indicating that it is not a fractional schedule.")
+      end
+
+      return annual_flh
+    end
+
+    # Returns an array of average hourly values from a ScheduleRuleset object
+    # Returns 8760 values, 8784 for leap years.
+    #
+    # @param schedule_ruleset [OpenStudio::Model::ScheduleRuleset] OpenStudio ScheduleRuleset object
+    # @return [Array<Double>] Array of hourly values for the year
+    def self.schedule_ruleset_get_hourly_values(schedule_ruleset)
+      # define the start and end date
```

Comment:
Since this is becoming a utility function, I wonder if we should add a safe guard here to make sure the argument provided is a ruleset schedule?

Reply:
added guard clauses to schedule_ruleset methods